### DISCUSSION TITLE
`result_callback` is ignored when only one subcommand exists

### DISCUSSION THREAD

**Discussion by multimeric** [fastapi/typer#1764] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Example Code

```python
import typer

app = typer.Typer(result_callback=lambda x: print(x))

@app.command()
def main() -> str:
    return "foo"

if __name__ == "__main__":
    app()
```

### Description

* Create a `Typed` instance
* Have only a single `@app.command()`
* Call the file on the CLI without using the command's name, ie `python test.py` rather than `python test.py main`
* The `result_callback` is not called. In the example above, that means nothing is printed to stdout.

### Operating System

Linux

### Operating System Details

Ubuntu 20.10

### Typer Version

0.6.1

### Python Version

3.9.5

**Comment by dilanmelvin** [comment]
Maintainers, could you please add the **bug** label?
This PR fixes Issue #445 where `result_callback` was ignored for single-command apps.
